geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/MemberConfigManager.java

```diff
@@ -77,68 +75,72 @@ public void delete(MemberConfig config, CacheConfig existing) {
     return generateMemberConfigs(memberInformation);
   }
 
+  @VisibleForTesting
+  Set<DistributedMember> getDistributedMembers(MemberConfig filter) {
+    Set<InternalDistributedMember> distributionManagerIds =
+        cache.getDistributionManager().getDistributionManagerIds();
+    if (filter.getId() != null) {
+      distributionManagerIds = distributionManagerIds.stream().filter(
+          internalDistributedMember -> (filter.getId().equals(internalDistributedMember.getName())))
+          .collect(Collectors.toSet());
+    }
+
+    return distributionManagerIds.stream()
+        .map(DistributedMember.class::cast).collect(Collectors.toSet());
+  }
+
```

Comment:
I prefer (because it is more efficient) to just conditionalize the stream, rather than add the extra steps of collecting and re-streaming, e.g. `Stream<foo> = foos.stream(); if (filter!=null) stream=stream.filter(...); return stream.collect(...)`

But there are enough times you have no choice but to collect and re-stream, I'm not going to object if you just prefer to always do it that way.